typescript/lib/networks/network_context.ts

```diff
@@ -0,0 +1,440 @@
+                )
+            ]
+        }
+
+        const enclaveDataVolMountDirpathOnSvcContainer: string = serviceResponse.getEnclaveDataVolumeMountDirpath();
+        if (enclaveDataVolMountDirpathOnSvcContainer == "") {
+            return [null, new Error(
+                "Kurtosis API reported an empty enclave data volume directory path for service " + serviceId + " - this should never happen, and is a bug with Kurtosis!",
+                )
+            ]
+        }
+
+        const serviceContext: ServiceContext = ServiceContext(
+            this.client,
+            serviceId,
+            serviceResponse.getIpAddr(),
+            this.enclaveDataVolMountpoint,
+            enclaveDataVolMountDirpathOnSvcContainer,
+        )
+
+        return [serviceContext, null]
+    }
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public removeService(serviceId: ServiceID, containerStopTimeoutSeconds: number): Error {
+
+        const log: logger.Logger = new logger.Logger();
+        log.debug("Removing service '%v'...", serviceId)
+        // NOTE: This is kinda weird - when we remove a service we can never get it back so having a container
+        //  stop timeout doesn't make much sense. It will make more sense when we can stop/start containers
+        // Independent of adding/removing them from the network
+        const args: RemoveServiceArgs = newRemoveServiceArgs(serviceId, containerStopTimeoutSeconds);
+        
+        // TODO TODO TODO - CALLBACK & ERROR-HANDLING
+        // if _, err := networkCtx.client.RemoveService(context.Background(), args); err != nil {
+        //     return stacktrace.Propagate(err, "An error occurred removing service '%v' from the network", serviceId)
+        // }
+
+        log.debug("Successfully removed service ID %v", serviceId)
+
+        return null;
+    }
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public repartitionNetwork(
+        partitionServices: Map<PartitionID, Map<ServiceID, boolean>>,
+        partitionConnections: Map<PartitionID, Map<PartitionID, PartitionConnectionInfo>>,
+        defaultConnection: PartitionConnectionInfo): Error {
+
+        if (partitionServices == null) {
+            return new Error("Partition services map cannot be nil");
+        }
+        if (defaultConnection == null) {
+            return new Error("Default connection cannot be nil");
+        }
+
+        // Cover for lazy/confused users
+        if (partitionConnections == null) {
+            partitionConnections = new Map();
+        }
+
+        const reqPartitionServices: Map<string, PartitionServices> = new Map();
+        for (let partitionId in partitionServices) {
+            const serviceIdSet = partitionServices[partitionId];
+
+            const serviceIdStrPseudoSet: Map<string, boolean> = new Map();
```

Comment:
This can be typed as a proper `Set` in Javascript, so we can actually rename the variable `serviceIdStrSet` (this being one of those examples where it's correct to deviate slightly from Go) and type it as `Set<string>`